### PR TITLE
[move-ide] Added support for name chain completions in attributes

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR adds support for handling name access chains in attributes. The goal here is NOT to add more general support smart auto-completion for attributes, though we can potentially do a lot of interesting things there. Instead, the focus here is to finish `::` auto-completion started in https://github.com/MystenLabs/sui/pull/18778

## Test plan 

All new and old tests must pass

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 1]
```diff
@@ -2461,6 +2491,10 @@ impl<'a> ParsingSymbolicator<'a> {
                         }
                     };
 
+                    fun.attributes.iter().for_each(|sp!(_, attrs)| {
+                        attrs.iter().for_each(|a| self.attr_symbols(a.clone()))
```
Comment (cgswords) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:2495]: Is there a reason we need the clone here? Can we rewrite `attr_symbols` to work over `&P::Attribute` and save ourselves some grief?

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 2]
```diff
@@ -2418,6 +2430,19 @@ impl<'a> ParsingSymbolicator<'a> {
...
+        use P::Attribute_ as A;
+        match attr {
+            A::Name(_) => (),
+            A::Assigned(_, v) => {
+                update_cursor!(self.cursor, *v, Attribute);
+            }
+            A::Parameterized(_, sp!(_, attributes)) => {
+                attributes.iter().for_each(|a| self.attr_symbols(a.clone()))
+            }
+        }
```
Comment (cgswords) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:2443]: I'm a little confused about this. In general, I would think we would call `update_cursor` at the _top_ of this function.

**Review by cgswords — APPROVED** [review]
Stamping so this can land, but with the understanding that the attribute tracking needs to be made more robust for known attribute autocompletions in the future.

**Comment by awelc** [comment]
> Stamping so this can land, but with the understanding that the attribute tracking needs to be made more robust for known attribute autocompletions in the future.

Thank you! To address both inline comments, for what I wanted to do here (just support the name access chains in attributes) this seemed simpler as it returns `AttributeValue` that can be processed right away, without recursively looking for it inside `Attribute`s. Totally agreed, though, that for a more general solution to smart attribute auto-completion, we'd need `CursorPosition` that marks `Attribute` instead of (just) `AttributeValue`

**Pull request merged**